zephyr/lib/alloc.c

```diff
@@ -257,8 +257,10 @@ static void virtual_heap_free(void *ptr)
 	ptr = (__sparse_force void *)sys_cache_cached_ptr_get(ptr);
 
 	ret = vmh_free(heap, ptr);
-	if (ret)
+	if (ret) {
 		tr_err(&zephyr_tr, "Unable to free %p! %d", ptr, ret);
+		k_panic();
```

Comment:
Do we really want to panic ? e.g. would a user space module be able to panic with a bad free() ? (and stop all other pipelines)

Reply:
> Do we really want to panic ? e.g. would a user space module be able to panic with a bad free() ? (and stop all other pipelines)

+1 - maybe we should add some notification to the host about similar important but not fatal errors

Reply:
failure of memory release:

- In the best situation it will lead to memory leak (and crash, sooner or later)
- in the worst - undefined actions because of cache inconsistency

In any case - we DO have a bug and it is a serious problem, and it is very hard to notice only by a line in the log. 
For me panic is fine here. OR - we may consider an assert that is checked in debug only. 

Anyway, SOF has no kernel that keeps the system working when an application crashes,  "user space module" is in fact a part of the system, not an application.